cpp/dolfin/la/PETScVector.h

```diff
@@ -126,21 +126,23 @@ class PETScVector
 
   /// Gather entries (given by global indices) into x.  This
   /// operation is collective
-  void gather(std::vector<double>& x,
+  void gather(std::vector<PetscScalar>& x,
               const std::vector<dolfin::la_index_t>& indices) const;
 
   /// Gather all entries into x on process 0.
   /// This operation is collective
-  void gather_on_zero(std::vector<double>& x) const;
+  void gather_on_zero(std::vector<PetscScalar>& x) const;
 
   /// Add multiple of given vector (AXPY operation)
-  void axpy(double a, const PETScVector& x);
+  void axpy(PetscScalar a, const PETScVector& x);
 
   /// Replace all entries in the vector by their absolute values
   void abs();
 
   /// Return dot product with given vector
-  double dot(const PETScVector& v) const;
+  /// For complex vectors, the argument v gets
+  /// complex conjugate
```

Comment:
Due to http://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Vec/VecDot.html the second argument gets the complex conjugate - which should be vector itself, not argument `v`. Double-check please.

Reply:
You're right. The way it is the vector itself gets conjugated.
However if by a.dot(b) we mean (a,b) the order of the arguments in VecDot should be changed.
For example, in petsc4py it's define that way.

Reply:
Changing the order of arguments makes sense to me. Real case wont be affected.